### PR TITLE
Timer-Recorder's should report 'transferSize' metric as response size

### PR DESCRIPTION
**Pull request by randolph-straub**

### CODE REVIEW

File: xlt-timerrecorder-chrome/src/background.js [hunk 1]
```diff
@@ -611,8 +613,10 @@ function createRecordEntry(dataEntry) {
   };
 
   const entriesCopy = {};
-  for (const u in dataEntry.entries) {
-    entriesCopy[u] = Array.from(dataEntry.entries[u]);
+  for (const i in dataEntry.entries) {
+    for (const u in dataEntry.entries[i]) {
+      entriesCopy[u] = Array.from(dataEntry.entries[i][u]);
+    }
   }
```
Comment (h-arlt) [inline xlt-timerrecorder-chrome/src/background.js:620]:
```suggestion
  (dataEntry.entries || []).forEach(function(eachEntry){
    // eachEntry maps URL strings (object keys) to an array of performance timing measurements
    for (const u in eachEntry) {
      entriesCopy[u] = Array.from(eachEntry[u]);
    }
  });
```

File: xlt-timerrecorder/background.js [hunk 2]
```diff
@@ -613,8 +615,10 @@ function createRecordEntry(dataEntry) {
   };
 
   const entriesCopy = {};
-  for (const u in dataEntry.entries) {
-    entriesCopy[u] = Array.from(dataEntry.entries[u]);
+  for (const i in dataEntry.entries) {
+    for (const u in dataEntry.entries[i]) {
+      entriesCopy[u] = Array.from(dataEntry.entries[i][u]);
+    }
   }
```
Comment (h-arlt) [inline xlt-timerrecorder/background.js:622]:
```suggestion
 (dataEntry.entries || []).forEach(function(eachEntry){
    // eachEntry maps URL strings (object keys) to an array of performance timing measurements
    for (const u in eachEntry) {
      entriesCopy[u] = Array.from(eachEntry[u]);
    }
  });
```

File: xlt-timerrecorder-chrome/src/page.js [hunk 3]
```diff
@@ -103,40 +103,17 @@ function createTimingData() {
...
-  newEntry.lastBytesTime = newEntry.firstBytesTime + newEntry.receiveTime;
-  newEntry.duration = newEntry.startTime && timing.responseEnd
-    ? timing.responseEnd - newEntry.startTime
-    : null;
-  newEntry.dnsTime = timing.domainLookupStart && timing.domainLookupEnd
-    ? timing.domainLookupEnd - timing.domainLookupStart
-    : null;
-
-  data[url] = [newEntry];
-
```
[the remark this answers is not in the document]
Comment (h-arlt) [reply xlt-timerrecorder-chrome/src/page.js:134]: Wait. The overall page timing is covered by `performance.getEntriesByType("navigation")` now, right? If so, please ignore my comment above.

[the remark this answers is not in the document]
Comment (randolph-straub) [reply xlt-timerrecorder-chrome/src/page.js:134]: Yes so far the first request was not part of the entries list and performance.timings does not have the transferSize. But now there is the navigation timing data which extends resource timing data and has all the required data we need for the first request. So now we can handle it like the other resource data.

**Pull request merged**